packages/financial-templates-lib/src/price-feed/CreatePriceFeed.js

```diff
@@ -174,6 +188,66 @@ async function createPriceFeed(logger, web3, networker, getTime, config) {
 
   // Internal helper methods:
 
+  // Returns an ExpressionPriceFeed.
+  async function _createExpressionPriceFeed(expressionConfig) {
+    // Build list of configs that could be used in the expression including default price feed configs and customFeeds
+    // that the user has provided inside the ExpressionPriceFeed config. Note: default configs are overriden by
+    // customFeeds with the same name. Tranform keys by escaping any special characters in the identifier names..
+    const allConfigs = Object.fromEntries(
+      Object.entries({ ...defaultConfigs, ...expressionConfig.customFeeds }).map(([key, value]) => {
+        return [escapeSpecialCharacters(key), value];
+      })
+    );
+
+    // This call chain:
+    // 1. Parses the expression into an expression tree of nodes.
+    // 2. Filters for "symbol" nodes, which would be price feed identifiers in this case.
+    // 3. Extract the name property for each of these symbol nodes
+    // 4. Puts it all in a set and converts back to an array to dedupe any repeated values.
+    const symbols = Array.from(
+      new Set(
+        math
+          .parse(expressionConfig.expression)
+          .filter(node => node.isSymbolNode)
+          .map(node => node.name)
+      )
+    );
+
+    // This is a complicated looking map that maps each symbol into an entry in an object with its value the price
+    // feed created from the mapped config in allConfigs.
+    const priceFeedMap = Object.fromEntries(
+      await Promise.all(
+        symbols.map(async symbol => {
+          const config = allConfigs[symbol];
+
+          // If there is no config for this symbol, insert null and send an error.
+          if (!config) {
+            logger.error({
```

Comment:
why doesnt this just throw an error, or log error and throw error?

Reply:
This is how the others work I believe -- just continuing the pattern from above where we log an error and return null.